src/main/java/edu/harvard/iq/dataverse/Shib.java

```diff
@@ -218,7 +218,26 @@ public void init() {
             ? getValueFromAssertion(shibAffiliationAttribute)
             : shibService.getAffiliation(shibIdp, shibService.getDevShibAccountType());
 
+
         if (affiliation != null) {
+            String ShibAffiliationSeparator = settingsService.getValueForKey(SettingsServiceBean.Key.ShibAffiliationSeparator);
+            if (ShibAffiliationSeparator == null) {
+                ShibAffiliationSeparator = ";";
+                }
+            String ShibAffiliationOrder = settingsService.getValueForKey(SettingsServiceBean.Key.ShibAffiliationOrder);
+            if (ShibAffiliationOrder != null) {
+                if (ShibAffiliationOrder.equals("lastAffiliation")) {
+                    affiliation = affiliation.substring(affiliation.lastIndexOf(ShibAffiliationSeparator) + 1); //patch for affiliation array returning last part
+                    }
+                else if (ShibAffiliationOrder.equals("firstAffiliation")) {
+                    try{
+                        affiliation = affiliation.substring(0,affiliation.indexOf(ShibAffiliationSeparator)); //patch for affiliation array returning first part
+                        }
+                    catch (Exception e){
+                        logger.info("Affiliation does not contains \"" + ShibAffiliationSeparator + "\"");
```

Comment:
All this code is strangely indented when I preview it in GitHub. Can you please use four spaces? Please see also https://guides.dataverse.org/en/5.11/developers/coding-style.html